trie/patriciaMerkleTrie_test.go

```diff
@@ -1537,6 +1537,379 @@ func TestPatriciaMerkleTrie_InsertOneValInNilTrie(t *testing.T) {
+			assert.True(t, found)
+			assert.Equal(t, []byte("reindeer"+strconv.Itoa(i)), val)
+		}
+
+		wg := sync.WaitGroup{}
+		wg.Add(numTrieValues + numBatchValues)
+		for i := 0; i < numTrieValues+numBatchValues; i++ {
+			go func(idx int) {
+				val, _, err := tr.Get([]byte("dog" + strconv.Itoa(idx)))
+				assert.Nil(t, err)
+				assert.Equal(t, []byte("reindeer"+strconv.Itoa(idx)), val)
+				wg.Done()
+			}(i)
+		}
+		wg.Wait()
+	})
+	t.Run("get multiple values from collapsed trie", func(t *testing.T) {
+		t.Parallel()
+
+		tr := emptyTrie()
+		numTrieValues := 1000
+		numBatchValues := 500
+		for i := 0; i < numTrieValues; i++ {
+			_ = tr.Update([]byte("dog"+strconv.Itoa(i)), []byte("reindeer"+strconv.Itoa(i)))
+		}
+		_ = tr.Commit()
+
+		// collapse the trie
+		rootHash, _ := tr.RootHash()
+		tr, _ = tr.Recreate(rootHash)
+
+		for i := numTrieValues; i < numTrieValues+numBatchValues; i++ {
+			_ = tr.Update([]byte("dog"+strconv.Itoa(i)), []byte("reindeer"+strconv.Itoa(i)))
+		}
+
+		// check some values are in the batch
+		bm := trie.GetBatchManager(tr)
+		for i := numTrieValues; i < numTrieValues+numBatchValues; i++ {
+			val, found := bm.Get(trie.KeyBytesToHex([]byte("dog" + strconv.Itoa(i))))
+			assert.True(t, found)
+			assert.Equal(t, []byte("reindeer"+strconv.Itoa(i)), val)
+		}
+
+		// get from collapsed trie
+		wg := sync.WaitGroup{}
+		wg.Add(numTrieValues + numBatchValues)
+		for i := 0; i < numTrieValues+numBatchValues; i++ {
+			go func(idx int) {
+				val, _, err := tr.Get([]byte("dog" + strconv.Itoa(idx)))
+				assert.Nil(t, err)
+				assert.Equal(t, []byte("reindeer"+strconv.Itoa(idx)), val)
+				wg.Done()
+			}(i)
+		}
+		wg.Wait()
+	})
+	t.Run("get value from batch that is currently being committed", func(t *testing.T) {
+		t.Parallel()
+
+		tr := emptyTrie()
+		waitForSignal := atomic.Bool{}
+		waitForSignal.Store(true)
+		startedProcessing := atomic.Bool{}
+		grm := &mock.GoroutinesManagerStub{
+			CanStartGoRoutineCalled: func() bool {
+				startedProcessing.Store(true)
+				return true
+			},
+			EndGoRoutineProcessingCalled: func() {
+				for waitForSignal.Load() {
+					time.Sleep(time.Millisecond * 100)
+				}
+			},
+		}
+		trie.SetGoRoutinesManager(tr, grm)
+
+		numOperations := 1000
+		for i := 0; i < numOperations; i++ {
+			_ = tr.Update([]byte("dog"+strconv.Itoa(i)), []byte("reindeer"))
+		}
+
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			trie.ExecuteUpdatesFromBatch(tr)
+			wg.Done()
+		}()
+
+		// wait for start of processing of the first batch
+		for !startedProcessing.Load() {
+			time.Sleep(time.Millisecond * 100)
+		}
+
+		for i := 0; i < numOperations; i++ {
+			val, _, err := tr.Get([]byte("dog" + strconv.Itoa(i)))
+			assert.Nil(t, err)
+			assert.Equal(t, []byte("reindeer"), val)
+		}
+
+		waitForSignal.Store(false)
+		// wait for end of processing of both batches
+		wg.Wait()
+	})
+}
+
+func TestPatricianMerkleTrie_ConcurrentOperations(t *testing.T) {
```

Comment:
add concurrency test with all operations in parallel

Reply:
or a TODO for this

Reply:
Will add for each PR the tests missing related to which trie func was refactored in that PR.